### PR TITLE
Ops 441 disable user accounts refactoring

### PR DESCRIPTION
**Pull request by johndeange**
## What changed

Added missing functionality to `/users` endpoint
Added USER_ADMIN role
Re-designed endpoint according to service layers.

## Issue

https://github.com/HHS/OPRE-OPS/issues/441

## How to test

Automated tests pass.

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [x] Form validations updated

### CODE REVIEW

**Comment by jonnalley** [comment]
@johndeange first off, 👏 🎉  but secondly, it seems at this point the notion of a "admin" wouldn't map to any known human at OPRE. Maybe the better thing to do is rename the current "admin" role the Budget Team

File: backend/data_tools/data/user_data.json5 [hunk 1]
```diff
@@ -137,8 +135,8 @@
         "POST_RESEARCH_PROJECT",
 
         "GET_USER",
-        "GET_USERS",
```
Comment (jonnalley) [inline backend/data_tools/data/user_data.json5:140]: am struggling to remember why we would have had GET_USERS (plural) in here

File: backend/data_tools/data/user_data.json5 [hunk 2]
```diff
@@ -218,6 +216,12 @@
         "GET_UPLOAD_DOCUMENT",
         "POST_UPLOAD_DOCUMENT"
       ]
+    },
+    { // 5 User Admin
+      name: "USER_ADMIN",
+      permissions: [
+        "POST_USER",
+      ],
```
Comment (jonnalley) [inline backend/data_tools/data/user_data.json5:224]: @johndeange so..... why not additional roles here?  I saw a few places in the code doing checks for `is_user_admin` on modifying or creating a user which seems to just be checking if a user has a role but it's not actually a check of the permission....    Is that by design?  I recognize there's a high chance of me asking out ignorance and/or out of context.

File: backend/data_tools/data/user_data.json5 [hunk 3]
```diff
@@ -137,8 +135,8 @@
         "POST_RESEARCH_PROJECT",
 
         "GET_USER",
-        "GET_USERS",
-
+        "PUT_USER",
+        "PATCH_USER",
```
Comment (jonnalley) [inline backend/data_tools/data/user_data.json5:139]: @johndeange Up to this point I had presumed that `USER` only affected the the same user as the one doing the verb whereas `USERS` would be something affecting a user other than themselves. I'm guessing that's not correct or at least not any longer. ?

Comment (johndeange) [reply backend/data_tools/data/user_data.json5:139]: The Permissions are currently all singular.  I think eventually `GET_USER` will allow access to the `GET` `/users/<id>` endpoint and `GET_USERS` will allow access to the `GET` `/users` endpoint.

**Comment by johndeange** [comment]
@jonnalley

Necessity of "admin" role - the idea here is the System Owner will be assigned the "admin" role and the "USER_ADMIN" would be delegated by the System Owner to a couple team members to create/update users.   I think there is still a need for a role for the Budget Team.

File: backend/data_tools/data/user_data.json5 (hunk 1 above)
Comment (johndeange) [reply backend/data_tools/data/user_data.json5:140]: See answer below

File: backend/data_tools/data/user_data.json5 (hunk 2 above)
Comment (jonnalley) [reply backend/data_tools/data/user_data.json5:224]: Ah, thank you. I'm better informed now :-)

**Review by maiyerlee — APPROVED** [review]
Looks good! Thanks for keeping the event type and initial data up to date. I like the concept of a safe user!

**Review by fpigeonjr — APPROVED** [review]
🟢 tests pass

**Pull request merged**